File: src/commands/customer_stats.py
import nextcord
from nextcord import Interaction, Embed
from nextcord.ext import commands
from src.database.supabase import supabase

bot = commands.Bot(command_prefix="!")

@bot.slash_command(name="customer-stats", description="Show your customer stats")
async def customer_stats(interaction: Interaction):
    user_id = str(interaction.user.id)

    # Fetch user stats from Supabase
    result = supabase.table("users").select("*").eq("id", user_id).execute()

    if not result.data:
        await interaction.response.send_message("❌ No stats found for you.", ephemeral=True)
        return

    user_data = result.data[0]
    points = user_data.get("points", 0)
    total_orders = user_data.get("total_orders", 0)

    embed = Embed(
        title=f"📊 Stats for {interaction.user.display_name}",
        description=f"Total Orders Completed: {total_orders}\nPoints: {points}",
        color=0x3498db
    )

    await interaction.response.send_message(embed=embed, ephemeral=True)

File: src/commands/customerStats.js
import { SlashCommandBuilder } from "discord.js";
import { supabase } from "../database/supabase.js";

export const customerStats = {
  data: new SlashCommandBuilder()
    .setName("customer-stats")
    .setDescription("Shows customer order stats"),

  async execute(interaction) {
    const userId = interaction.user.id;

    // Fetch stats from Supabase
    const { data: stats, error } = await supabase
      .from("orders")
      .select("*", { count: "exact" })
      .eq("customer_id", userId)
      .eq("status", "completed");

    if (error) return interaction.reply({ content: "Failed to fetch stats.", ephemeral: true });

    const points = stats.length; // assuming 1 point per order
    await interaction.reply({
      content: `📊 You have completed ${stats.length} orders and earned ${points} points.`
    });
  }
};
